src/Manifest/DefaultManifestGenerator.php

```diff
@@ -24,61 +27,136 @@ public function __construct(MetadataProvider $metadataProvider, ManifestSerializ
+        ManifestOptions $manifestOptions,
+        QueryMetadata $queryMetadata,
+    ): void {
+        $columns = $queryMetadata->getColumns();
+        $schema = [];
+
+        foreach ($columns as $column) {
+            $schema[] = $this->createSchemaEntry($column, $column->isPrimaryKey());
+        }
+
+        $manifestOptions->setSchema($schema);
+    }
+
+    private function generateSchema(Table $table, ?array $columns, ?array $primaryKeys): array
+    {
+        $schema = [];
         $allTableColumns = $table->getColumns();
-        $columnMetadata = [];
-        $sanitizedPks = [];
-        $exportedColumns = $exportConfig->hasColumns() ?
-            $exportConfig->getColumns() : $allTableColumns->getNames();
+        $exportedColumns = $columns ?: $allTableColumns->getNames();
 
-        foreach ($exportedColumns as $index => $columnName) {
+        foreach ($exportedColumns as $columnName) {
             $column = $allTableColumns->getByName($columnName);
-            $columnMetadata[$column->getSanitizedName()] = $this->serializer->serializeColumn($column);
+            $isPrimaryKey = $primaryKeys && in_array($column->getName(), $primaryKeys, true);
+            $backend = $table->hasDatatypeBackend() ? $table->getDatatypeBackend() : null;
+            $schema[] = $this->createSchemaEntry($column, $isPrimaryKey, $backend);
+        }
 
-            // Sanitize PKs defined in the configuration
-            if ($exportConfig->hasPrimaryKey() &&
-                in_array($column->getName(), $exportConfig->getPrimaryKey(), true)
-            ) {
-                $sanitizedPks[] = $column->getSanitizedName();
-            }
+        return $schema;
+    }
+
+    private function createSchemaEntry(
+        Column $column,
+        bool $isPrimaryKey,
+        ?string $backend = null,
+    ): ManifestOptionsSchema {
+        $metadata = [
+            'KBC.sourceName' => $column->getName(),
+            'KBC.sanitizedName' => $column->getSanitizedName(),
+            'KBC.uniqueKey' => $column->isUniqueKey(),
+            'KBC.ordinalPosition' => $column->hasOrdinalPosition() ? $column->getOrdinalPosition() : null,
+            'KBC.autoIncrement' => $column->isAutoIncrement() ?: null,
+            'KBC.autoIncrementValue' => $column->hasAutoIncrementValue() ? $column->getAutoIncrementValue() : null,
+        ];
+
+        if ($column->hasForeignKey()) {
+            $fk = $column->getForeignKey();
+            $metadata['KBC.foreignKey'] = true;
+            $metadata['KBC.foreignKeyName'] = $fk->hasName() ? $fk->getName() : null;
+            $metadata['KBC.foreignKeyRefSchema'] = $fk->hasRefSchema() ? $fk->getRefSchema() : null;
+            $metadata['KBC.foreignKeyRefTable'] = $fk->getRefTable();
+            $metadata['KBC.foreignKeyRefColumn'] = $fk->getRefColumn();
+        }
+
+        foreach ($column->getConstraints() as $constraint) {
+            $metadata['KBC.constraintName'] = $constraint;
```

Comment:
Toto funguje jak když máš víc constraints? Jeden vládne všem?

Reply:
hmmm, mrknu

Reply:
Ale taky tam je tenhle problém. To si ještě dneska otočím na tom syncu... Dobrej point